packages/rps/src/utils/channelClient.ts

```diff
@@ -0,0 +1,219 @@
+  jsonrpc: '2.0';
+  id: string;
+  result: ResultType;
+}
+
+export interface CreateChannelParameters extends UpdateChannelParameters {
+  appDefinition: string;
+}
+
+export interface ChannelResult extends CreateChannelParameters {
+  channelId: string;
+  status: ChannelStatus;
+  funding: Funding[];
+}
+
+export interface CreateChannelRequest
+  extends JsonRPCRequest<CreateChannelParameters, MethodName.CreateChannel> {}
+
+export interface CreateChannelResult extends ChannelResult {
+  turnNum: 0;
+}
+
+export interface CreateChannelResponse extends JsonRPCResponse<ChannelResult> {}
+
+export interface JoinChannelParameters {
+  channelId: string;
+}
+
+export interface JoinChannelRequest
+  extends JsonRPCRequest<JoinChannelParameters, MethodName.JoinChannel> {}
+
+export interface JoinChannelResult extends ChannelResult {
+  turnNum: 1;
+}
+
+export interface JoinChannelResponse extends JsonRPCResponse<ChannelResult> {}
+
+export interface UpdateChannelParameters {
+  participants: Participant[];
+  allocations: Allocation[];
+  appData: string;
+}
+
+export interface UpdateChannelRequest
+  extends JsonRPCRequest<UpdateChannelParameters, MethodName.UpdateChannel> {}
+
+export interface UpdateChannelResult extends ChannelResult {
+  turnNum: number;
+}
+
+export interface ChannelProposedNotification
+  extends JsonRPCNotification<CreateChannelResult, 'ChannelProposed'> {}
+
+export interface ChannelUpdatedNotification
+  extends JsonRPCNotification<UpdateChannelResult, 'ChannelUpdated'> {}
+
+export type ActionParameters =
+  | CreateChannelParameters
+  | JoinChannelParameters
+  | UpdateChannelParameters;
+
+export class ChannelClient {
+  protected events = new EventEmitter();
+
+  onMessageReceived(
+    notificationName: NotificationName,
+    callback: (parameters: ActionParameters) => void
+  ) {
+    this.events.on(
+      'message',
+      (message: JsonRPCNotification<ActionParameters> | JsonRPCRequest<ActionParameters>) => {
+        if (message.method === notificationName) {
+          callback(message.params);
+        }
+      }
+    );
+  }
+
+  createChannel(parameters: CreateChannelParameters) {
+    this.send(MethodName.CreateChannel, parameters);
+
+    // TODO: This notification payload is incomplete, since it doesn't have the result
+    // of the channel creation. Where should this go?
+    this.notifyChannelProposed(parameters as CreateChannelResult);
+  }
+
+  joinChannel(parameters: JoinChannelParameters) {
+    this.send(MethodName.JoinChannel, parameters);
+
+    // TODO: This notification payload is incomplete, since it doesn't have the result
+    // of the channel creation. Where should this go?
+    this.notifyChannelUpdated(parameters as UpdateChannelResult);
+  }
+
+  updateChannel(parameters: UpdateChannelParameters) {
+    this.send(MethodName.UpdateChannel, parameters);
+  }
+
+  protected async send(methodName: MethodName, parameters: ActionParameters) {

```

Comment:
I was finding it a little confusing in the code to figure out what `send` meant. Maybe calling this `sendToWallet` or something would help with this?

Reply:
Sure!